src/main/java/StringCalc/StringCalculator.java

```diff
@@ -0,0 +1,52 @@
+package StringCalc;
+
+import java.util.regex.Pattern;
+
+public class StringCalculator {
+
+    public int add(String input) {
+        if (input == null || input.trim().isEmpty()) {
+            return 0;
+        }
+
+        String delimiter = ",|:";
+        String numbers = input;
+
+        if (input.startsWith("//")) {
+            String[] parts = input.split("\n", 2);
+            delimiter = Pattern.quote(parts[0].substring(2));
+            numbers = parts[1];
+        }
+
+        String[] tokens = numbers.split(delimiter);
+        int sum = 0;
+
+        StringBuilder negatives = new StringBuilder();
+
+        for (String token : tokens) {
+            if (token.trim().isEmpty()) {
+                continue;
+            }
+
+            try {
+                int number = Integer.parseInt(token.trim());
+                if (number < 0) {
+                    if (negatives.length() > 0) {
+                        negatives.append(", ");
+                    }
+                    negatives.append(number);
```

Comment:
number가 0 미만이면 바로 예외를 throw 해도 되지 않을까요? 현재의 negatives StringBuilder 사용으로도 물론 결과를 도출할 수 있지만 불필요한 동작이 늘어난 것 같아요

```suggestion
                if (number < 0) {
                    throw new RuntimeException("음수는 안 됩니다. 입력된 값: " + negatives);
                }
```

Reply:
어떠한 음수 값을 입력했는지 보여주고 싶어서 저 코드를 작성하였는데, 의미가 없을 것 같기도 합니다. 따라서, 말씀해주신 것 처럼 바로 예외처리를 진행하는 방향으로 수정하였습니다.